File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
    prev:=head
    current:=head
    hashset:=map[int]bool{}
    for current!=nil{
        if _,ok:=hashset[current.Val];ok{
            prev.Next=current.Next
        }else{
            hashset[current.Val]=true
            prev=current
        }
        current=current.Next
    }
    return head
}

File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    let hashset=new Set();
    let prev=new ListNode(0);
    let current=head;
    while(current){
        if(hashset.has(current.val)){
            prev.next=current.next;
        }
        else{
            prev=current;
            hashset.add(current.val);
        }
        current=current.next
    }
    return head;
};